include/system_info.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "dense_map.h"
+#include "types.h"
+
+#include <bitset>
+
```

Comment:
I noticed that `host_memory_id` and `user_memory_id` are currently defined in `types.h`. Semantically I feel like it might make more sense to define them in this file, but this is not a big deal either way and if there are any dependency constraints preventing that then ignore this point.

Reply:
I agree. `types.h` is kind of a kitchen sink anyway. It defines a lot of types that are not conceptually related simply because they are all integer aliases.

Reply:
True, but `host_memory_id` etc. aren't even types!